docs/input/docs/getting-started/principles.md

```diff
@@ -20,13 +20,84 @@ according to the user-made registrations.
 | Conversion    | `IConvert<in TSourceClass, in TTargetClass, TConvertIntention>`| `IBaseConvertIntention` |
 | Copy          | `ICopy<in TClassToCopy>`                                       | no intention            |
 
-### Sub operations
+## Registration interfaces
 
 The paragraph above states the technical difference between the operations regarding the business case specific intention.
 This is a rather minor difference - As mentioned before, the base objects do the job or resolving the sub operations
 defined in user-code by the responsible developer on a defined structure.
-Basically, one registration is needed for each data class type for a specific intention. The registrations
+One registration is needed for each data class type for a specific intention. The registrations
 can be seen as walking through the tree and deciding for each node which properties and child / parent objects need
 to be handled in a certain way which can be done by using what is generally called a _sub operation_. These
 _sub operations_ are specific to each of the operation, though some can be mixed (for example, while converting an object implementing
 the same interface as the master data it is being converted from the interface's properties can be copied using a sub copy).
+
+There is one registration interface per each operation supported by the library. All of these registration
+interfaces share the same raison d'être and work as containers which hold the definition for each operation and
+can retrieved from the IoC container.
+
+| Operation     | Technical                                                     |
+| ------------- |--------------------------------------------------------------:|
+| Comparison    | `ICompareRegistrations<TModelToCompare, TCompareIntention>`   |
+| Conversion    | `IConvertRegistrations<TSource, TTarget, TConvertIntention>`  |
+| Copy          | `ICopyRegistrations<T>`                                       |
+
+### Note: Registration interfaces
```

Comment:
Format this as an annotation block, for example (see [Bootstrap alerts](https://getbootstrap.com/docs/4.0/components/alerts/) for possible alert classes):

```
:::{.alert .alert-info}
Consider these steps while implementing registration interfaces:

* Register your implementation of the registration in the IoC container.
* Do not traverse over multiple tree leafs when registering properties, except for reference level sub operations.
* It's usually better to unit test each registration, not the entire tree being copied.
:::
```